src/makeInfinityAdder.js

```diff
@@ -6,7 +6,27 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+  let emptyArgCall = false;
+
+  const addition = function(number) {
+    if (arguments.length === 0) {
+      if (!emptyArgCall) {
+        emptyArgCall = true;
+
+        return sum;
+      }
```

Comment:
```suggestion
  const addition = function(number) {
    if (arguments.length === 0) {
      const result = sum;

      sum = 0;

      return result;
    }
```

also works and looks simpler but it's up to you

Reply:
Agree let's use this advice